### PR TITLE
WIP: Datatable converters and refactored buffers & storers

### PR DESCRIPTION
**Pull request by nokome**
A PR which implements converters for `Datatables`. 

Also refactoring of converters, buffers and storers based on experiences with other implementing other converters and integration in environments other than Stencila Desktop (i.e. in `stencila/node` for containers). More rationale for this to come. Implementing these as new classes so as not to break existing APIs used for documents.

### CODE REVIEW

File: make.js [hunk 1]
```diff
@@ -171,21 +173,27 @@ function buildEnv() {
 }
 
 // reads all fixtures from /tests/ and writes them into a script
-function buildTestBackend() {
-  b.custom('Creating test backend...', {
-    src: './tests/documents/**/*',
-    dest: './tmp/test-vfs.js',
+function buildTestFixtures() {
```
Comment (nokome) [inline make.js]: Renaming to 'fixtures'

File: make.js [hunk 2]
```diff
@@ -171,21 +173,27 @@ function buildEnv() {
...
-        let content = fs.readFileSync(f).toString()
-        let relPath = path.relative(rootDir, f).replace(/\\/g, '/')
-        vfs[relPath] = content
+      const fixtures = {}
+      files.forEach((file) => {
+        if (b.isDirectory(file)) return
+        let ext = path.extname(file)
+        let data = fs.readFileSync(file)
+        
+        // Base64 encode non-text files
```
Comment (nokome) [inline make.js]: Non-textish files are getting base64 encoded

File: src/datatable/DatatableODSConverter.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+   * 
+   * @param  {string} content - XML string of the `content.xml` file
+   * @return {DOMElement}
+   */
+  _importDatatableFromContent(content) {
+    let {$datatable, $$} = this._importCreateElement()
+    let $fields = $$('fields')
+    let $values = $$('values')
+
+    // This suceeds in the Node.js but in the browser fails to parse
```
Comment (nokome) [inline src/datatable/DatatableODSConverter.js]: @oliver---- : there seems to be an issue parsing XML with namespaced tagName (?)

File: src/datatable/DatatableODSConverter.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    // maybe dure to namespaced tag names e.g. `office:document-content`
+    let $content = DefaultDOMElement.parseXML(content)
+
+    // Using a CSS selector here. e.g
+    //    let $table = $content.find('office:body office:spreadsheet table:table')
+    // does not work because namespaced tag names are not supported
+    // Is there a better way to do this?
```
Comment (nokome) [inline src/datatable/DatatableODSConverter.js]: oliver---- : any thoughts here?

Comment (oliver----) [inline src/datatable/DatatableODSConverter.js]: I need to investigate. Need to find a consistent way for browser and memory DOM.

**Comment by nokome** [comment]
Moved to https://github.com/stencila/convert/pull/7

**Pull request closed without merging**